### PR TITLE
Add support for `WithSummary` and `WithDescription` metadata

### PR DESCRIPTION
**Pull request by hwoodiwiss**
fixes https://github.com/dotnet/aspnetcore/issues/40753
fixes https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2548

Adds support for the MinimalAPI operation metadata functions, `WithSummary` and `WithDescription`, including CI changes.

### CODE REVIEW

**Comment by hwoodiwiss** [comment]
@domaindrivendev Just pinging for a review now that we're at .NET 7 RC1.
Also looking for some thoughts, I know this has been superseded a bit by WithOpenApi, but it might still be nice to support the new metadata piecemeal?

I'm happy to either expand this to support the rest of the new Metadata types, leave it as is, or scrap it altogether. Just looking for your thoughts on this.

**Comment by esskar** [comment]
Hi, why is this still open?

**Comment by hwoodiwiss** [comment]
I'd honestly forgotten about this. Ideally I think this should still be reviewed and merged, a quick check indicates that support for WithSummary and WIthDescription hasn't been added yet. (I'll test that out a bit later just in case)
If this is still needed, I'll update it from main, and look to get a review.

**Comment by DES-Destry** [comment]
Hello, will this code merged? I want it to use already :)

**Comment by cutpix** [comment]
Can we finally merge it?

**Comment by hwoodiwiss** [comment]
> Can we finally merge it?

Based on https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2759#issuecomment-1936136992, it looks like this repo is on hiatus until a new maintainer can be found.

Until then, it's unlikely this will be merged. It looks like this PR is being ported over to a fork of this project, so it could be worth checking that out, or checking out [NSwag] (https://github.com/RicoSuter/NSwag).

**Comment by martincostello** [comment]
@hwoodiwiss If you rebase on to the default branch to pick up our new CI we can take a look at this for you.

File: test/Swashbuckle.AspNetCore.SwaggerGen.Test/Swashbuckle.AspNetCore.SwaggerGen.Test.csproj
Comment (hwoodiwiss) [inline test/Swashbuckle.AspNetCore.SwaggerGen.Test/Swashbuckle.AspNetCore.SwaggerGen.Test.csproj:6]: Not 100% sure how this will reconcile with other changes in the backlog, should this multitarget and the tests that require .NET 7 or greater be `#ifdef`'d?

Comment (martincostello) [reply test/Swashbuckle.AspNetCore.SwaggerGen.Test/Swashbuckle.AspNetCore.SwaggerGen.Test.csproj:6]: Good question, I'm not sure. For now maybe `TargetFrameworks` would be the more conservative approach?

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:662]: Maybe it would be neater to `#if` the two property assignments and the entire declaration of these methods instead? The net result should be the same for .NET <7 that the two properties have a value of `null`?

**Review by martincostello — APPROVED** [review]

**Pull request merged**
